HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,343 @@
+    {
+        throw "Invalid matrix dimensions. Expected 2D.";
+    }
+
+    Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+    for(int i = 0; i < cvMat.rows; i++)
+    {
+        for(int j = 0; j < cvMat.cols; j++)
+        {
+            eigenMat(i, j) = cvMat.at<double>(i, j);
+        }
+    }
+
+    return eigenMat;
+}
+
+Representations transformFromZividToRobot(const Eigen::Matrix3d &rotationMatrix)
+{
+    Representations retVals;
+    // Rotation Matrix in Transformation Matrix to Axis-angle
+    std::cout << "\nConverting rotation matrix to axis-angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle = Eigen::AngleAxisd(rotationMatrix);
+    retVals.angleAxis = reorderAxisAngle(axisAngle);
+    std::cout << "\nAxis-angle:\n" << retVals.angleAxis << std::endl;
+
+    // Axis-angle to Rotation Vector
+    std::cout << "\nConverting axis-angle to rotation vector" << std::endl;
+    if(!axisAngle.axis().isUnitary())
+    {
+        std::cout << "Note! Axis in axis-angle pair is not unitary" << std::endl;
+    }
+    retVals.rotationVector = axisAngle.angle() * axisAngle.axis();
+    std::cout << "\nRotationVector:\n" << retVals.rotationVector << std::endl;
+
+    // Rotation Matrix to Quaternion
+    std::cout << "\nConverting rotation matrix to quaternion" << std::endl;
+    const Eigen::Quaterniond quaternion = Eigen::Quaterniond(rotationMatrix);
+    retVals.quaternion = quaternion.coeffs();
+    std::cout << "\nQuaternion:\n" << retVals.quaternion << std::endl;
+
+    // Rotation Matrix to Roll-Pitch-Yaw
+    retVals.rotations = { { { RotationConvention::ZYX_Intrinsic, "ZYX_Intrinsic", "RollPitchYawIntrinsicZYX" },
+                            { RotationConvention::XYZ_Extrinsic, "XYZ_Extrinsic", "RollPitchYawExtrinsicXYZ" },
+                            { RotationConvention::XYZ_Intrinsic, "XYZ_Intrinsic", "RollPitchYawIntrinsicXYZ" },
+                            { RotationConvention::ZYX_Extrinsic, "ZYX_Extrinsic", "RollPitchYawExtrinsicZYX" } } };
+    for(int i = 0; i < nofRotConv; i++)
+    {
+        std::cout << "\nConverting from rotation matrix to roll-pitch-yaw angles (" << retVals.rotations[i].shortDesc
+                  << ")" << std::endl;
+        auto temp_rollPitchYaw = rotmat2rpy(rotationMatrix, retVals.rotations[i].convention);
+        std::cout << "\nRollPitchYaw" << retVals.rotations[i].shortDesc << ":\n" << temp_rollPitchYaw << std::endl;
+        retVals.rotations[i].rollPitchYaw = temp_rollPitchYaw;
+    }
+
+    return retVals;
+}
+
+void transformFromRobotToZivid(const Eigen::Matrix3d &rotationMatrix, const Representations &representations)
+{
+    // Saving conversion to .YAML file
+    auto fileStorageOut = cv::FileStorage::FileStorage();
+    if(!fileStorageOut.open("robotTransform.yaml", cv::FileStorage::Mode::WRITE))
+    {
+        std::cout << "Could not open file for writing" << std::endl;
+    }
+
+    // Roll, Pitch and Yaw to Rotation Matrix
+    for(int i = 0; i < nofRotConv; i++)
```

Comment:
This for loop would be simpler if you did `for(const auto rotation : representations.rotations)` instead